src/util/Cache.h

```diff
@@ -227,6 +226,37 @@ class FlexibleCache {
     // TODO<joka921>:: implement this functionality
   }
 
+  MemorySize getMaxSizeSingleEntry() const noexcept {
+    return _maxSizeSingleEntry;
+  }
+
+  void recomputeSize(const Key& key) {
+    // Pinned entries must not be dynamic in nature
+    AD_CONTRACT_CHECK(!containsPinned(key));
+    if (!containsNonPinned(key)) {
+      return;
+    }
+    auto newSize = _valueSizeGetter(*(*this)[key]);
+    auto& oldSize = _sizeMap.at(key);
+    // Entry has grown too big to completely keep within the cache or we can't
+    // fit it in the cache
+    if (_maxSizeSingleEntry < newSize ||
+        _maxSize - std::min(_totalSizePinned, _maxSize) < newSize) {
```

Comment:
This I don't understand/believe:

why not
```
sizeDiff = newSize - oldSize;  // check for underflow...
if (totalSizePinnedPlusNonPinnedEverything + sizeDiff > _maxSize) erase(key);
```.
(+ The `_maxSizeSingleEntry` check of course, that one is undisputed.

Reply:
Because `_maxSize` is of type `MemorySize` which will throw an exception instead of becoming a negative size value. I remember fiddling around with this a lot until it worked and this is what my changes converged to. I must admit that I don't know anymore why I settled on these exact changes.